cmd/ci-operator-configresolver/main.go

```diff
@@ -128,6 +130,9 @@ func validateOptions(o options) error {
 	if o.validateOnly && o.flatRegistry {
 		return errors.New("--validate-only and --flat-registry flags cannot be set simultaneously")
 	}
+	if o.configMapReloader && !o.flatRegistry {
+		return errors.New("If cm-reloader is set, flat-registry must also be set")
```

Comment:
Similarly, couldn't we default `--flat-registry` if we detect we're reading from a configmap?

Reply:
I assume so, but I prefer to be more explicit about the flags/args instead of silently enabling them. Either way would work.